### PR TITLE
sysusers: only check whether the requested GID is available

### PR DESCRIPTION
**Pull request by lucab**
This relaxes the availability check when creating a group, if an
explicit GID has been requested.
It avoids mixing up users and groups entries with valid and unique
UIDs/GIDs, but each having the same ID number.

Fixes: https://github.com/systemd/systemd/issues/24066

### CODE REVIEW

**Comment by lucab** [comment]
I did cumulative manual testing of the rollup of this PR plus https://github.com/systemd/systemd/pull/24249 against the whole set of users/groups on a Fedora system (manually translated by me for now, hopefully without human errors), and I didn't spot any other apparent issues.

File: src/sysusers/sysusers.c [hunk 1]
```diff
@@ -1169,6 +1169,28 @@ static int add_user(Item *i) {
         return 0;
 }
 
+static int gid_is_available(gid_t gid) {
```
Comment (poettering) [inline src/sysusers/sysusers.c]: hmm, so uid_is_ok() has this boolean parameter `check_with_gid`, which allows controlling whether the uid check should also imply a gid check. It appears we should do the inverse here, and thus modify gid_is_ok() to take a paramter `which_with_uid`.

Comment (lucab) [reply src/sysusers/sysusers.c]: Ah good point, indeed that keeps it symmetrical. I guess parameter name suggestion was a typo for `check_with_uid`.

Comment (poettering) [reply src/sysusers/sysusers.c]: yupp, it was. sorry for the confusion

Comment (lucab) [reply src/sysusers/sysusers.c]: No problem, it was clearly understandable from the context. Reworked the patch that way.

**Pull request merged**